server/channels/store/sqlstore/sqlx_wrapper.go

```diff
@@ -459,3 +470,45 @@ func printArgs(query string, dur time.Duration, args ...any) {
 	}
 	mlog.Debug(query, fields...)
 }
+
+func (w *sqlxDBWrapper) checkErrWithResult(res sql.Result, err error) (sql.Result, error) {
+	return res, w.checkErr(err)
+}
+
+func (w *sqlxDBWrapper) checkErrWithRows(res *sqlx.Rows, err error) (*sqlx.Rows, error) {
+	return res, w.checkErr(err)
+}
+
+func (w *sqlxDBWrapper) checkErr(err error) error {
+	var netError *net.OpError
+	if errors.As(err, &netError) && (!netError.Temporary() && !netError.Timeout()) {
+		w.isOnline.Store(false)
```

Comment:
I wanted to keep a single `return err` statement. Otherwise, technically you can return after any of the `w.isOnline.Store(false)` statements, and then it gets complicated with named returns and such.